nexus/reconfigurator/planning/src/blueprint_editor/sled_editor/zones.rs

```diff
@@ -65,13 +64,14 @@ impl ZonesEditor {
         self.counts
     }
 
-    pub fn zones(
+    pub fn zones<F>(
         &self,
-        filter: BlueprintZoneFilter,
-    ) -> impl Iterator<Item = &BlueprintZoneConfig> {
-        self.zones
-            .iter()
-            .filter(move |config| config.disposition.matches(filter))
+        mut filter: F,
+    ) -> impl Iterator<Item = &BlueprintZoneConfig>
+    where
+        F: FnMut(BlueprintZoneDisposition) -> bool,
```

Comment:
Why does this have to be `FnMut`. Doesn't `Fn` fit better for a filter?

Reply:
`FnMut` is more general; I copied the style from https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter. All the current callers would certainly be fine with `Fn`. There miiiight be some use for a test to pass an FnMut (e.g., to accumulate a count or something?)?